plonky_block_proof_gen/src/lib.rs

```diff
@@ -90,8 +90,41 @@
 //!     curr_block_agg_proof: &GeneratedAggProof,
 //! ) -> ProofGenResult<GeneratedBlockProof> { ... }
 //! ```
+//!
+//! ## Verifying block proofs
+//!
+//! The `ProverState` can be used to verify any block proofs emitted with the
+//! same set of circuits.
+//! However, because the prover state can be quite heavy, the necessary verifier
+//! data to verify block proofs can be saved independently into a
+//! `VerifierState`, to allow anyone to easily verify block proofs.
+//!
+//! ```compile_fail
+//!     # use plonky_block_proof_gen::prover_state::ProverStateBuilder;
+//!     # use plonky_block_proof_gen::verifier_state::VerifierState;
+//!     let mut builder = ProverStateBuilder::default();
+//!
+//!     // Generate a `ProverState` from the builder.
+//!     let prover_state = builder.build();
+//!
+//!     // Derive a `VerifierState` from the `ProverState`.
+//!     let verifier_state: VerifierState = prover_state.into();
+//!
+//!     // The prover generates some block proof.
+//!     let block_proof = prover_state.generate_block_proof(...);
+//!     
+//!     // Have the verifier attest validity of the proof.
+//!     assert!(verifier_state.verify(block_proof.intern).is_ok());
+//! ```
```

Comment:
Alright I need to start doing doc comments like this in my PRs. :slightly_smiling_face: